src/main/java/de/tum/in/www1/artemis/config/Constants.java

```diff
@@ -54,6 +54,10 @@
 
     public static final Pattern SHORT_NAME_PATTERN = Pattern.compile(SHORT_NAME_REGEX);
 
+    public static final String FILE_PATTERN_REGEX = "(doc|DOC|docx|DOCX|dotx|DOTX|xml|XML|pdf|PDF|wps|WPS|wpd|WPD|wp|WP|wp4|WP4|wp5|WP5|wp6|WP6|wp7|WP7|qxd|QXD|ps|PS|pub|PUB|tex|TEX|vsd|VSD|bmp|BMP|jpg|JPG|jpeg|JPEG|pjpeg|PJEPG|gif|GIF|pgn|PGN|png|PNG|txt|TXT|rtf|RTF|wav|WAV|mp3|MP3|html|HTML|odt|ODT|xls|XLS|xlsx|XLSX|wks|WKS|xlr|XLR|csv|CSV|ppt|PPT|pps|PPS|ppsx|PPXS)$";
```

Comment:
This is quite an extensive list of file formats. However we are missing some archiving formats, markdown, epub... I am just thinking here if we need a strict validation? I guess either we provide a small list of possible extensions and then we can show them in the hint or in the error message when creating file upload exercise or we allow any extensions that has 2-4 numbers and/or characters.

Reply:
Thank you for the feedback! 

I was considering that as well and there are two reasons why I decided upon this validation: 
1) Until now tutors specified extensions such as "not allowed" and if for some reason they intend to keep this misusing the file uploads with this use case of creating exercises where students are not supposed to upload anything, it will still remain a possibility for them to do so - they will still be able to specify stuff like "none" and my PR will not have really fulfilled its purpose. 
2) Stephan suggested that I do not limit the list too much. 

As I now realize, I missed some important formats, so one option would be to simply extend the list further. 

Let's see what the other reviewers think as well and I will discuss this tomorrow at the end of the meeting. We can think of the most common use cases and make sure that the extensions for them are allowed. ^^

Reply:
Sounds good. One more suggestion from my side here is to do the check case insensitive. It is fine if the user types `DoCx` for example, since we lower case everything afterwards.